internal/cmd/gocdk/provision.go

```diff
@@ -54,12 +60,170 @@ func registerProvisionCmd(ctx context.Context, pctx *processContext, rootCmd *co
 	rootCmd.AddCommand(provisionCmd)
 }
 
+const provisionStaticRootDir = "/provision"
+
 // The "provision list" command.
-func provisionList(pctx *processContext) {
-	pctx.Logf("NOT YET IMPLEMENTED")
+func provisionList(pctx *processContext) error {
+	// TODO(rvangent): There must be an easier way to get a listing... add
+	// a helper to static.
+	dir, err := static.Open(provisionStaticRootDir)
+	if err != nil {
+		return xerrors.Errorf("provision list: couldn't open static resources: %w", err)
+	}
+	ptInfos, err := dir.Readdir(-1)
+	if err != nil {
+		return xerrors.Errorf("provision list: couldn't read static resources: %w", err)
+	}
+	// The entries at this level are portable APIs, e.g., "blob".
+	for _, ptInfo := range ptInfos {
+		ptName := ptInfo.Name()
+		if !ptInfo.IsDir() {
+			continue
+		}
+		ptTypeDir, err := static.Open(path.Join(provisionStaticRootDir, ptName))
+		if err != nil {
+			return xerrors.Errorf("provision list: couldn't open static resources: %w", err)
+		}
+		providerInfos, err := ptTypeDir.Readdir(-1)
+		if err != nil {
+			return xerrors.Errorf("provision list: couldn't read static resources: %w", err)
+		}
+		for _, info := range providerInfos {
+			if !info.IsDir() {
+				continue
+			}
+			pctx.Println(path.Join(ptName, info.Name()))
+		}
+	}
+	return nil
 }
 
 // The "provision add" command.
+// TODO(rvangent): Can we support adding a particular type more than once?
+// TODO(rvangent): If things fail in the middle, we are in an undefined state.
+//                 Unclear how to handle that....
+// TODO(rvangent): Modifying Terraform files in place means that we need to run
+//                 "terraform init" again; currently we don't.
+// TODO(rvangent): Currently all of the variables.tf files have default values, so
+//                 "terraform apply" works fine. Add the ability to prompt the user
+//                 for things (e.g., GCP project ID), and then instantiate auto.tfvars
+//                 files with the values.
 func provisionAdd(ctx context.Context, pctx *processContext, biome, typ string) error {
-	return errors.New("not yet implemented")
+	pctx.Logf("Adding %q to %q...", typ, biome)
+
+	typeParts := strings.Split(typ, "/")
+	if len(typeParts) != 2 {
+		return xerrors.Errorf("provision add: %q is not a supported type; use 'gocdk provision list' to see available types", typ)
+	}
+
+	moduleDir, err := pctx.ModuleRoot(ctx)
+	if err != nil {
+		return xerrors.Errorf("provision add: %w", err)
+	}
+	dstPath := biomeDir(moduleDir, biome)
+
+	srcRoot := path.Join(provisionStaticRootDir, typ)
```

Comment:
I am concerned that a user could specify a string that allows them to use a different directory inside the static resources. Consider using an explicit allowlist.

Reply:
Added a TODO. This will be easier when the other TODO about a helper for reading the dirs is done.